CMakeLists.txt

```diff
@@ -174,9 +176,52 @@ foreach(src_file ${uhdm-GENERATED_SRC})
   set_source_files_properties(${src_file} PROPERTIES GENERATED TRUE)
 endforeach(src_file ${uhdm-GENERATED_SRC})
 
+# Parse out version and configure header
+if(EXISTS "${CMAKE_SOURCE_DIR}/.git")
+  # running a build from the git repo, parse out tag and sha
+  # get latest commit
+  execute_process(COMMAND git describe --tags --abbrev=0
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    OUTPUT_VARIABLE LATEST_VERSION)
+
+  # parse out first number after "v"
+  string(REGEX REPLACE "^v([0-9]+)\\..*" "\\1" UHDM_VERSION_MAJOR "${LATEST_VERSION}")
+  # parse out second number, e.g. after "v[0-9]."
+  string(REGEX REPLACE "^v[0-9]+\\.([0-9]+).*" "\\1" UHDM_VERSION_MINOR "${LATEST_VERSION}")
+
+  # get latest commit
+  execute_process(COMMAND git rev-parse HEAD
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    OUTPUT_VARIABLE UHDM_VERSION_COMMIT_SHA)
+  # strip newline
+  string(REGEX REPLACE "\n$" "" UHDM_VERSION_COMMIT_SHA "${UHDM_VERSION_COMMIT_SHA}")
+
+  if("${UHDM_VERSION_MAJOR}" STREQUAL "")
+    # No tags locally, user project version
+    set(UHDM_VERSION_MAJOR ${CMAKE_PROJECT_VERSION_MAJOR})
+    set(UHDM_VERSION_MINOR ${CMAKE_PROJECT_VERSION_MINOR})
+  elseif(NOT "${UHDM_VERSION_MAJOR}.${UHDM_VERSION_MINOR}" STREQUAL "${CMAKE_PROJECT_VERSION_MAJOR}.${CMAKE_PROJECT_VERSION_MINOR}")
+    message(FATAL_ERROR "Git tag ${UHDM_VERSION_MAJOR}.${UHDM_VERSION_MINOR} \
+      does not match project version ${CMAKE_PROJECT_VERSION_MAJOR}.${CMAKE_PROJECT_VERSION_MINOR}. \
+      Did you forget to update project version in CMakeLists.txt?")
+  endif()
+else()
+  set(UHDM_VERSION_MAJOR ${CMAKE_PROJECT_VERSION_MAJOR})
+  set(UHDM_VERSION_MINOR ${CMAKE_PROJECT_VERSION_MINOR})
+  set(UHDM_VERSION_COMMIT_SHA "release")
+endif()
+
+set(UHDM_BUILD_TYPE ${CMAKE_BUILD_TYPE})
+message("Building version v${UHDM_VERSION_MAJOR}.${UHDM_VERSION_MINOR} [${UHDM_VERSION_COMMIT_SHA}]")
+configure_file(${CMAKE_SOURCE_DIR}/include/uhdm-version.h.in ${GENDIR}/uhdm/uhdm-version.h)
+
 add_library(uhdm ${uhdm-GENERATED_SRC})
 set_target_properties(uhdm PROPERTIES PUBLIC_HEADER ${GENDIR}/uhdm/uhdm.h)
 
+if(BUILD_SHARED_LIBS)
+  set_property(TARGET uhdm PROPERTY POSITION_INDEPENDENT_CODE 1)
```

Comment:
Is there a way to set the soname version, so that we end up with libuhdm.so.1.48 as well as symbolic links libuhdm.so.1 and libuhdm.so in an install target ?